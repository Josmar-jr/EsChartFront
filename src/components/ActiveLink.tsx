import Link, { LinkProps } from 'next/link';
import { useRouter } from 'next/router';
import { cloneElement, ReactElement } from 'react';

type ActiveLink = LinkProps & {
  children: ReactElement;
  shouldMatchExactHref?: boolean;
};

const afterStyle =
  "after:content-[''] after:absolute after:h-[1.5px] after:w-full after:rounded-r after:block after:top-[35px] after:left-[-1px]";

export function ActiveLink({
  children,
  shouldMatchExactHref = false,
  ...rest
}: ActiveLink) {
  const { asPath } = useRouter();

  let isActive = false;

  if (shouldMatchExactHref && (asPath === rest.href || asPath === rest.as)) {
    isActive = true;
  }

  if (
    !shouldMatchExactHref &&
    (asPath.startsWith(String(rest.href)) || asPath.startsWith(String(rest.as)))
  ) {
    isActive = true;
  }

  return (
    <Link {...rest}>
      {cloneElement(children, {
        className: `text-sm py-2 hover:bg-gray-200 transition-colors rounded-md transition-all focus:ring focus:secondary outline-none inline-block px-[12px] w-full py-1 relative ${afterStyle} ${
          isActive ? 'text-primary font-bold dark:after:bg-secondary dark:text-secondary after:bg-primary' : 'text-slate-500 dark:text-slate-400'
        }`
      })}
    </Link>
  );
}
