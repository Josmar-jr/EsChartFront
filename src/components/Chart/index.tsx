import {
  CartesianGrid,
  Legend,
  Line,
  LineChart,
  Tooltip,
  XAxis,
  YAxis
} from 'recharts';

import { format } from 'date-fns';
import enUS from 'date-fns/locale/en-US';

import { CustomTooltip } from './CustomTooltip';

const data = [
  {
    date: format(new Date('02/24/2022'), 'dd/MM/yyyy', {
      locale: enUS
    }),
    pv: 2400
  },
  {
    date: new Date('01/24/2022').toString(),
    pv: 1398
  },
  {
    date: new Date('11/14/2022').toString(),
    pv: 9800
  },
  {
    date: new Date().toString(),
    pv: 3908
  },
  {
    date: new Date('06/16/2022').toString(),
    pv: 4800
  }
];

export function Chart() {
  return (
    <LineChart
      width={800}
      height={400}
      data={data}
      className="bg-blue-200 bg-opacity-20 rounded-md p-4"
      margin={{
        top: 5,
        right: 30,
        left: 20,
        bottom: 5
      }}
    >
      <CartesianGrid strokeDasharray="6 6" stroke="#d4d4d8" />
      <XAxis dataKey="date" format="Y-m-d" />
      <YAxis />
      <Tooltip
        offset={-90}
        content={<CustomTooltip />}
        coordinate={{
          x: 0,
          y: 0
        }}
      />
      <Legend />
      <Line
        type="monotone"
        dataKey="pv"
        stroke="#0095db"
        strokeWidth={3}
        activeDot={{ r: 6 }}
        dot={false}
      />
    </LineChart>
  );
}
