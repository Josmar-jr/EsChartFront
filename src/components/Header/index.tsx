import { List } from 'phosphor-react';
import { useDisclosure } from '../../contexts/DisclosureContext';
import { Menu } from './Menu';

export function Header() {
  const { onOpen } = useDisclosure();

  return (
    <header className="flex justify-between bg-neutral dark:bg-slate-900 border-b fixed top-0 left-0 border-slate-300 dark:border-slate-800 w-full py-2 px-4">
      <button
        onClick={onOpen}
        className="text-primary dark:text-secondary outline-none rounded-sm hover:brightness-95 focus:brightness-95 focus:ring"
      >
        <List size={26} />
      </button>
      <Menu />
    </header>
  );
}
