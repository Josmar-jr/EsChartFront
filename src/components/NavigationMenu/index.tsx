import * as NavMenuPrimitive from '@radix-ui/react-navigation-menu';
import { NavLink } from './NavLink';

export function NavigationMenu() {
  return (
    <NavMenuPrimitive.Root className="max-w-7xl mx-auto flex relative w-full pb-2">
      <NavMenuPrimitive.List className="flex justify-center items-center">
        <NavMenuPrimitive.Item>
          <NavMenuPrimitive.Trigger />
          <NavMenuPrimitive.Content>
            <NavMenuPrimitive.Link />
          </NavMenuPrimitive.Content>
        </NavMenuPrimitive.Item>

        <NavMenuPrimitive.Item>
          <NavLink href="/dashboard">Dashboard</NavLink>
        </NavMenuPrimitive.Item>
        <NavMenuPrimitive.Item>
          <NavLink href="/register">Register</NavLink>
        </NavMenuPrimitive.Item>
        <NavMenuPrimitive.Item>
          <NavLink href="/me/as">Settings</NavLink>
        </NavMenuPrimitive.Item>
      </NavMenuPrimitive.List>

      <NavMenuPrimitive.Viewport />
    </NavMenuPrimitive.Root>
  );
}
