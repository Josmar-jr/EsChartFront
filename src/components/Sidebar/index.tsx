import { ChartPieSlice, Gear, SquaresFour } from 'phosphor-react';
import { NavLink } from './NavLink';

export function Sidebar() {
  return (
    <aside className="border border-red-600">
      <nav>
        <ul className="flex gap-5 flex-col text-slate-500 font-medium">
          <li>
            <a href="#" className="flex">
              <SquaresFour size={20} />
              Dashboard
            </a>
          </li>
          <li>
            <NavLink icon={<ChartPieSlice size={20} />} href="/dashboard">
              Statistics
            </NavLink>
          </li>
          <li>
            <a href="#" className="flex">
              <Gear size={20} />
              Settings
            </a>
          </li>
        </ul>
      </nav>
    </aside>
  );
}
