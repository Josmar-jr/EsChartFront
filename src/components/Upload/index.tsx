import { FolderOpen } from 'phosphor-react';
import Dropzone from 'react-dropzone';

export function Upload({ onUpload }) {
  function renderDragMessage(isDragActive: boolean, isDragReject: boolean) {
    if (!isDragActive) {
      return <p>Arraste arquivos aqui...</p>;
    }

    if (isDragReject) {
      return <p>Arquivo n√£o suportado</p>;
    }

    return <p>Solte os arquivos aqui</p>;
  }

  return (
    <Dropzone
      accept={{
        'text/csv': [
          '.csv',
          '.xls',
          '.xlsx',
          'text/csv',
          'application/csv',
          'text/comma-separated-values',
          'application/csv',
          'application/excel',
          'application/vnd.msexcel',
          'text/anytext',
          'application/vnd.ms-excel',
          'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
        ]
      }}
      onDropAccepted={onUpload}
    >
      {({ getRootProps, getInputProps, isDragActive, isDragReject }) => (
        <div
          {...getRootProps()}
          className={`${isDragActive && 'outline-slate-500'} ${
            isDragReject && 'outline-red-400'
          } ${
            isDragActive && 'bg-blueOpacity'
          } outline-primary outline-dashed text-center outline-2 outline-offset-2 rounded-md cursor-pointer transition-[height] duration-200 ease-in`}
        >
          <input type="file" {...getInputProps()} />
          <div className="flex flex-col text-slate-500 items-center justify-center py-16">
            <FolderOpen size={36} />
            {renderDragMessage(isDragActive, isDragReject)}
          </div>
        </div>
      )}
    </Dropzone>
  );
}
