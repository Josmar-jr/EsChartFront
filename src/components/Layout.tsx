import { ReactNode } from 'react';
import { DisclosureProvider } from '../contexts/DisclosureContext';
import { Header } from './Header';
import { Sidebar } from './Sidebar';

interface LayoutProps {
  children: ReactNode;
}

export function Layout({ children }: LayoutProps) {
  return (
    <>
      <DisclosureProvider>
        <>
          <Sidebar />
          <Header />
        </>
      </DisclosureProvider>
      <main className="mt-16 ml-4 md:ml-6">{children}</main>
    </>
  );
}
