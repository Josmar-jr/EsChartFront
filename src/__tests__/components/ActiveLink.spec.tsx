import { render, screen, cleanup } from '@testing-library/react';
import { ActiveLink } from '../../components/ActiveLink';
import renderer from 'react-test-renderer';

jest.mock('next/router', () => {
  return {
    useRouter() {
      return {
        asPath: '/dashboard'
      };
    }
  };
});

describe('Sidebar Component', () => {
  it('should render correctly', () => {
    render(
      <ActiveLink href="/dashboard">
        <a>Dashboard</a>
      </ActiveLink>
    );

    const link = screen.getByText('Dashboard');

    expect(link).toBeInTheDocument();
  });

  it('should add class if the link is currently active', () => {
    render(
      <ActiveLink href="/dashboard">
        <a>Dashboard</a>
      </ActiveLink>
    );

    const link = screen.getByText('Dashboard');

    expect(link).toHaveClass('text-primary after:bg-primary');
  });

  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <ActiveLink href="/dashboard">
          <a>Dashboard</a>
        </ActiveLink>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
