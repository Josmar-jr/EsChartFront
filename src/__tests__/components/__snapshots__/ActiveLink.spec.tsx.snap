// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component should match snapshot 1`] = `
<a
  className="hover:text-primary transition-all focus:ring focus:secondary outline-none inline-block px-[12px] w-full py-1 relative after:content-[''] after:absolute after:h-10 after:w-[3.8px] after:rounded-r after:block after:top-[-4px] after:left-[-1px] text-primary after:bg-primary"
  href="/dashboard"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  Dashboard
</a>
`;
